sampling.go

```diff
@@ -0,0 +1,303 @@
+		return err
+	}
+	if err := validateParentBasedComponent(p.RemoteSampled); err != nil {
+		return err
+	}
+	if err := validateParentBasedComponent(p.RemoteNotSampled); err != nil {
+		return err
+	}
+	if err := validateParentBasedComponent(p.LocalSampled); err != nil {
+		return err
+	}
+	return validateParentBasedComponent(p.LocalNotSampled)
+}
+
+func getSamplerConfig(s Sampler) (*sampling.Config, error) {
+	if s == nil {
+		return nil, nil
+	}
+	config, err := s.convert()
+	if err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+func (p ParentBased) convert() (*sampling.Config, error) {
+	pbc := sampling.DefaultParentBasedSampler()
+	samplers := make(map[sampling.SamplerID]sampling.SamplerConfig)
+	rootSampler, err := getSamplerConfig(p.Root)
+	if err != nil {
+		return nil, err
+	}
+	if rootSampler != nil {
+		pbc.Root = rootSampler.ActiveSampler
+		for id, config := range rootSampler.Samplers {
+			if config.Config != nil {
+				samplers[id] = config
+			}
+		}
+	}
+
+	remoteSampledSampler, err := getSamplerConfig(p.RemoteSampled)
+	if err != nil {
+		return nil, err
+	}
+	if remoteSampledSampler != nil {
+		pbc.RemoteSampled = remoteSampledSampler.ActiveSampler
+		for id, config := range remoteSampledSampler.Samplers {
+			if config.Config != nil {
+				samplers[id] = config
+			}
+		}
+	}
+
+	remoteNotSampledSampler, err := getSamplerConfig(p.RemoteNotSampled)
+	if err != nil {
+		return nil, err
+	}
+	if remoteNotSampledSampler != nil {
+		pbc.RemoteNotSampled = remoteNotSampledSampler.ActiveSampler
+		for id, config := range remoteNotSampledSampler.Samplers {
+			if config.Config != nil {
+				samplers[id] = config
+			}
+		}
+	}
+
+	localSampledSamplers, err := getSamplerConfig(p.LocalSampled)
+	if err != nil {
+		return nil, err
+	}
+	if localSampledSamplers != nil {
+		pbc.LocalSampled = localSampledSamplers.ActiveSampler
+		for id, config := range localSampledSamplers.Samplers {
+			if config.Config != nil {
+				samplers[id] = config
+			}
+		}
+	}
+
+	localNotSampledSampler, err := getSamplerConfig(p.LocalNotSampled)
+	if err != nil {
+		return nil, err
+	}
+	if localNotSampledSampler != nil {
+		pbc.LocalNotSampled = localNotSampledSampler.ActiveSampler
+		for id, config := range localNotSampledSampler.Samplers {
+			if config.Config != nil {
+				samplers[id] = config
+			}
+		}
+	}
+
+	samplers[sampling.ParentBasedID] = sampling.SamplerConfig{
+		SamplerType: sampling.SamplerParentBased,
+		Config:      pbc,
+	}
+
+	return &sampling.Config{
+		Samplers:      samplers,
+		ActiveSampler: sampling.ParentBasedID,
+	}, nil
+}
+
+func newSamplerFromEnv() (Sampler, error) {
+	defaultSampler := ParentBased{
+		Root:             AlwaysOn{},
+		RemoteSampled:    AlwaysOn{},
+		RemoteNotSampled: AlwaysOff{},
+		LocalSampled:     AlwaysOn{},
+		LocalNotSampled:  AlwaysOff{},
+	}
+
+	samplerName, ok := lookupEnv(tracesSamplerKey)
+	if !ok {
+		return defaultSampler, nil
+	}
```

Comment:
To match the other environment variable resolution functionality, and to not perform wasted work, the default sampler should not be returned when the sampler environment variable is not provided.

```suggestion
	samplerName, ok := lookupEnv(tracesSamplerKey)
	if !ok {
		return nil, nil
	}
	
	defaultSampler := ParentBased{
		Root:             AlwaysOn{},
		RemoteSampled:    AlwaysOn{},
		RemoteNotSampled: AlwaysOff{},
		LocalSampled:     AlwaysOn{},
		LocalNotSampled:  AlwaysOff{},
	}
```

Reply:
Updated